### PR TITLE
Update Docker instructions

### PR DESCRIPTION
**Pull request by Kaixhin**
The CUDA Docker images are moving to use the NVIDIA Docker program, so I've updated the docs appropriately.

### CODE REVIEW

**Comment by nouiz** [comment]
Just a note that if you want GPU docker with Theano, there is some there:

https://github.com/nouiz/Theano-Docker

File: docs/user/installation.rst [hunk 1]
```diff
@@ -228,21 +228,21 @@ To check whether it is found by Theano, run the following command:
...
-Docker images
-=============
+Docker
+======
 
-Instead of manually installing Theano and Lasagne on your machines as described
-above, you may want to use a pre-made `Docker <https://www.docker.com/whatisdocker>`_
-image: `Lasagne Docker (CPU) <https://hub.docker.com/r/kaixhin/lasagne/>`_ or
+Builds of Lasagne are available as `Docker <https://www.docker.com>`_
+images: `Lasagne Docker (CPU) <https://hub.docker.com/r/kaixhin/lasagne/>`_ or
```
Comment (f0k) [inline docs/user/installation.rst]: Why this change? The previous sentence and section title made explicit that this is an alternative way of installing/using Theano and Lasagne.

Comment (Kaixhin) [inline docs/user/installation.rst]: With other docs I wrote in this more formal way (see Theano for example). Now that you mention it the reason must be that I literally took your wording from https://github.com/Lasagne/Lasagne/issues/411. Would you like to have the wording changed back or not? I can just amend this commit to keep it tidy.

Comment (f0k) [inline docs/user/installation.rst]: > Now that you mention it the reason must be that I literally took your wording from #411.

Ah, that must be why I liked the original wording better ;)

> Would you like to have the wording changed back or not? I can just amend this commit to keep it tidy.

Yes, amending the comment to restore the original formulation would be appreciated! Also I think the direct link to "what is docker" is more useful for people unfamiliar with it than just linking to docker.com. The latter is easy to find from the former, but not vice versa!

Comment (Kaixhin) [inline docs/user/installation.rst]: OK will do. In terms of the link, it's actually changed from https://www.docker.com/whatisdocker to https://www.docker.com/what-docker, so even though there's a redirect the URL seems like it could keep changing (unlike the top-level website). But I'll put in the new address as they will hopefully keep redirects in place.

**Comment by f0k** [comment]
I think you forgot the `git add` before `git commit --amend` and `git push --force`.

**Comment by f0k** [comment]
Thank you, merging!

**Comment by benanne** [comment]
Thanks :)
